### PR TITLE
fix: redesign transaction history

### PR DESCRIPTION
**Pull request by tiankii**
This pull request introduces comprehensive improvements and new tests for date and transaction badge utilities, focusing on enhancing date formatting logic, increasing test coverage, and improving test reliability. The changes include new and improved tests for date utilities, adjustments to transaction date formatting logic, and various test infrastructure enhancements.

**Date utilities and formatting improvements:**

- Added a new test suite for `@app/utils/date` covering `isSameDay`, `isToday`, `isYesterday`, and `formatShortDate`, including edge cases like leap years and timezone handling.
- Updated `@app/components/transaction-date` tests to mock `isToday` and `isYesterday`, ensuring that relative vs. absolute date formatting is tested accurately and language-agnostic.

**Transaction history and screen test enhancements:**

- Added a new test suite for the `TransactionHistoryScreen` to verify correct grouping and formatting of transaction dates (relative for recent, short date for older), including realistic GraphQL mocks and timezone handling.
- Improved the reliability of tests involving `InteractionManager.runAfterInteractions` by mocking its implementation to run synchronously. [[1]](diffhunk://#diff-e24c8758660acaa1f94a72d419ea025bb35a1739461942bdfe2474686c43b4f5R2) [[2]](diffhunk://#diff-e24c8758660acaa1f94a72d419ea025bb35a1739461942bdfe2474686c43b4f5R21-R30) [[3]](diffhunk://#diff-dbd72197536e992df78a06116174b7540571840445f59a1b48590d98b4a7bcecR1-R279)
- Simplified navigation test in the home screen test suite by removing unnecessary parameters from the expectation.

**Badge and hook testing improvements:**

- Added a comprehensive test suite for the `useOutgoingBadgeVisibility` hook, covering visibility logic, timer handling, and cleanup on unmount or transaction ID change.
- Updated mocks for feature flags and Apollo client in unseen transaction badge tests to improve test isolation and reliability.

**Animation and mock updates:**

- Updated a jest mock to use `useDropInOutAnimation` instead of `useDropInAnimation` in unseen transaction amount badge tests, aligning with the current component implementation.

These changes collectively improve the robustness, reliability, and coverage of the test suite for date and transaction-related UI components and utilities.

### CODE REVIEW

File: __tests__/screens/transaction-history-dates.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+import { RootStackParamList } from "@app/navigation/stack-param-lists"
+import { TransactionHistoryScreen } from "@app/screens/transaction-history"
+import {
+  TransactionListForDefaultAccountDocument,
+  TxLastSeenDocument,
+} from "@app/graphql/generated"
+import { loadLocale } from "@app/i18n/i18n-util.sync"
+
+import { ContextForScreen } from "./helper"
+import { clear } from "@app/utils/storage"
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply __tests__/screens/transaction-history-dates.spec.tsx]: Done!

File: app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }, 50)
+
+    const hideTimeout = setTimeout(() => {
+      setVisible(false)
+      onHide?.()
+    }, ttlMs + 50)
+
+    return () => {
+      clearTimeout(showTimeout)
+      clearTimeout(hideTimeout)
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts]: Done!

```diff
diff --git a/app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts b/app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts
index 7da92ab18..3877552b8 100644
--- a/app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts
+++ b/app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts
@@ -21,18 +21,21 @@ export const useOutgoingBadgeVisibility = ({
       return
     }

+    let hideTimeout: ReturnType<typeof setTimeout> | undefined
+
     const showTimeout = setTimeout(() => {
       setVisible(true)
+      hideTimeout = setTimeout(() => {
+        setVisible(false)
+        onHide?.()
+      }, ttlMs)
     }, 50)

-    const hideTimeout = setTimeout(() => {
-      setVisible(false)
-      onHide?.()
-    }, ttlMs + 50)
-
     return () => {
       clearTimeout(showTimeout)
-      clearTimeout(hideTimeout)
+      if (hideTimeout !== undefined) {
+        clearTimeout(hideTimeout)
+      }
     }
   }, [txId, isOutgoing, amountText, ttlMs, onHide])
```

File: app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  }, [txId, isOutgoing, amountText, ttlMs, onHide])
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts:40]: This is intentional. The `onHide` callback changes when `latestUnseenTx` changes, which is the same trigger for the effect to re-run anyway (via `txId`). So there's no extra re-execution happening

File: app/navigation/root-navigator.tsx [hunk 4]
```diff
@@ -350,7 +351,12 @@ export const RootStack = () => {
       <RootNavigator.Screen
         name="transactionHistory"
         component={TransactionHistoryScreen}
-        options={{ title: LL.TransactionScreen.transactionHistoryTitle() }}
+        options={{
+          title: LL.TransactionScreen.transactionHistoryTitle(),
+          presentation: "modal",
+          cardStyleInterpolator: CardStyleInterpolators.forVerticalIOS,
+          gestureEnabled: false,
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/navigation/root-navigator.tsx:358]: The default value of `gestureEnabled` is `true`. I've disabled it intentionally to prevent users from dismissing this screen by swiping down. This is required behavior for this modal

File: app/screens/home-screen/home-screen.tsx [hunk 5]
```diff
@@ -265,6 +266,17 @@ export const HomeScreen: React.FC = () => {
...
 
+  const showOutgoingBadge = useOutgoingBadgeVisibility({
+    txId: latestUnseenTx?.id,
+    amountText: unseenAmountText,
+    isOutgoing,
+    onHide: () => {
+      if (latestUnseenTx?.settlementCurrency) {
+        markTxSeen(latestUnseenTx.settlementCurrency)
+      }
+    },
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/home-screen/home-screen.tsx]: Done! Wrapped the callback in `useCallback` to prevent unnecessary effect re-runs

File: app/screens/transaction-history/transaction-history-screen.tsx [hunk 6]
```diff
@@ -344,6 +387,14 @@ export const TransactionHistoryScreen: React.FC<TransactionHistoryScreenProps> =
 
 const useStyles = makeStyles(({ colors }) => ({
   loadingContainer: { justifyContent: "center", alignItems: "center", flex: 1 },
+  skeletonWrapper: { flex: 1, alignSelf: "stretch" },
+  skeletonContainer: { alignSelf: "stretch" },
+  loaderBackground: {
+    color: colors.loaderBackground,
+  },
+  loaderForefound: {
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/transaction-history/transaction-history-screen.tsx]: Done! Fixed the typo

File: app/screens/transaction-history/transaction-history-screen.tsx [hunk 7]
```diff
@@ -344,6 +387,14 @@ export const TransactionHistoryScreen: React.FC<TransactionHistoryScreenProps> =
...
+    color: colors.loaderForeground,
+  },
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/transaction-history/transaction-history-screen.tsx]: Done! Removed unused styles

File: app/components/header-back-control/header-back-control.tsx [hunk 8]
```diff
@@ -22,7 +22,14 @@ type HeaderBackControlParams = {
...
 export const headerBackControl = ({ canGoBack = true }: HeaderBackControlParams = {}) =>
-  canGoBack ? undefined : () => <InvisibleBackButton />
+  canGoBack
+    ? (props: React.ComponentProps<typeof HeaderBackButton>) => {
+        const {
+          theme: { colors },
+        } = useTheme()
+        return <HeaderBackButton {...props} pressColor={colors.grey5} pressOpacity={1} />
+      }
+    : () => <InvisibleBackButton />
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/header-back-control/header-back-control.tsx]: Done! Extracted the anonymous function to a named component to keep a stable reference.

```diff
+const HeaderBackButtonWithTheme: React.FC<
+  React.ComponentProps<typeof HeaderBackButton>
+> = (props) => {
+  const {
+    theme: { colors },
+  } = useTheme()
+  return <HeaderBackButton {...props} pressColor={colors.grey5} pressOpacity={1} />
+}
+
 export const headerBackControl = ({ canGoBack = true }: HeaderBackControlParams = {}) =>
-  canGoBack
-    ? (props: React.ComponentProps<typeof HeaderBackButton>) => {
-        const {
-          theme: { colors },
-        } = useTheme()
-        return <HeaderBackButton {...props} pressColor={colors.grey5} pressOpacity={1} />
-      }
-    : () => <InvisibleBackButton />
+  canGoBack ? HeaderBackButtonWithTheme : InvisibleBackButton
```

File: app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    return () => {
+      clearTimeout(showTimeout)
+      if (hideTimeout !== undefined) {
+        clearTimeout(hideTimeout)
+      }
+    }
+  }, [txId, isOutgoing, amountText, ttlMs, onHide])
+
+  return visible
+}
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/unseen-tx-amount-badge/use-outgoing-badge-visibility.ts:43]: Done! Added tests: `__tests__/hooks/use-outgoing-badge-visibility.spec.ts`

File: app/screens/transaction-history/transaction-history-skeleton.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+import ContentLoader, { Rect } from "react-content-loader/native"
+import { makeStyles } from "@rn-vui/themed"
+
+const TransactionHistorySkeleton = () => {
+  const styles = useStyles()
+
+  return (
+    <View style={styles.container}>
+      <ContentLoader
+        height={640}
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/transaction-history/transaction-history-skeleton.tsx]: Done! Now using `useWindowDimensions` for responsive height.

File: app/screens/transaction-history/transaction-history-screen.tsx [hunk 11]
```diff
@@ -344,6 +387,7 @@ export const TransactionHistoryScreen: React.FC<TransactionHistoryScreenProps> =
 
 const useStyles = makeStyles(({ colors }) => ({
   loadingContainer: { justifyContent: "center", alignItems: "center", flex: 1 },
+  skeletonWrapper: { flex: 1, alignSelf: "stretch" },
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/transaction-history/transaction-history-screen.tsx:389]: Done! Removed unused `loadingContainer` style.
